lib/faktory_worker/connection.ex

```diff
@@ -21,28 +21,45 @@ defmodule FaktoryWorker.Connection do
     password = Keyword.get(opts, :password)
 
     with {:ok, connection} <- socket_handler.connect(host, port, opts),
-         :ok <- verify_handshake(connection, password) do
+         {:ok, _} <- verify_handshake(connection, password) do
       {:ok, connection}
     end
   end
 
   @spec send_command(connection :: Connection.t(), Protocol.protocol_command()) ::
-          :ok | {:error, term()}
+          {:ok, String.t()} | {:error, term()}
   def send_command(%{socket_handler: socket_handler} = connection, command) do
-    case Protocol.encode_command(command) do
-      {:ok, payload} -> socket_handler.send(connection, payload)
-      {:error, _} = error -> error
+    with {:ok, payload} <- Protocol.encode_command(command),
+         :ok <- socket_handler.send(connection, payload),
```

Comment:
Should this be getting a tuple back for consistency?

Reply:
I don't think it should be since there is no return value. I believe this is the general convention used in Elixir.